src/requestManager.h

```diff
@@ -63,14 +64,20 @@ class CUnknownObj
 class CRequestManager
 {
   protected:
-  // map of transactions
+
+  // keeps track of the order of block requests so we can iterate through
+  // the mapBlkInfo in the order that block INV or HEADERS were received.
+  // This keeps blocks from returning out of order.
```

Comment:
out of order blocks can and do cause unnecessary timeouts and re-requests during IBD.  Imagine you've just request 8 large blocks from a node, with the youngest block last...now you have to wait for all 8 blocks to download before you can connect any of them...now what if that node is also a slow downloader ...now you're stuck, when you could have been doing useful work , until you get a timeout and end up re-requesting the blocks from another node.   This doesn't solve all the problems to do with out of order blocks such as when two blocks are mined closely in time and inv's/headers show up out of order, however, it does solve the problem most often seen.